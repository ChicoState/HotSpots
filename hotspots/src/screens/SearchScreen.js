import React, { useState } from "react";
import { View, Text, StyleSheet } from "react-native";
import SearchBar from "../components/SearchBar";
import yelp from "../api/yelp";

const SearchScreen = function() {
  /*
   * The useState() hook takes the default state as an argument and returns an array
   * with a stateful value (term) and a function to update it (setTerm).
   * The SearchBar component below uses this to update (term)
   * which allows us to pass through user input wherever we need within the project.
   */
  const [term, setTerm] = useState("");
  // This is another piece of state which is used to store GET requests from the Yelp API.
  const [businesses, setBusinesses] = useState([]);
  // State that is used to display error messages.
  const [errorMessage, setErrorMessage] = useState("");
  /*
   * This is an async/await pattern which requests data using GET
   * and assigns the data sent back to us in const response.
   * Notice the async and await declarations prepended to the functions.
   */
  const searchApi = async function() {
    try {
      const response = await yelp.get("/search", {
        params: {
          limit: 50,
          term,
          location: "chico"
        }
      });
      // Businesses state is updated using the setBusinesses function.
      setBusinesses(response.data.businesses);
    } catch (err) {
      setErrorMessage("Something went wrong");
    }
  };

  return (
    <View>
      <SearchBar term={term} onTermChange={setTerm} onTermSubmit={searchApi} />
      {errorMessage ? <Text>{errorMessage}</Text> : null}
      <Text>We have found {businesses.length} results</Text>
    </View>
  );
};

const styles = StyleSheet.create({});

export default SearchScreen;
