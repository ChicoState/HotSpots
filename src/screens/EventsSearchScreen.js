import React, {useState} from 'react';
import { View, Text, TextInput, StyleSheet } from 'react-native';

const EventsSearchScreen = () => {
    const [searchTerm, setSearchTerm] = useState('')
    const [city, setCity] = useState('')
    const [date, setDate] = useState('')

    return (
        <View>
            <Text>Search For Upcoming Events</Text>
            <View style={styles.entryStyle}>
                <Text style={styles.labelStyle}>*Event: </Text>
                <TextInput
                   style={styles.inputStyle} 
                   placeholder='Select Event'
                   autoCorrect={false}
                   value={searchTerm}
                   onChangeText={setSearchTerm}
                />

            </View>
            <View style={styles.entryStyle}>
                <Text style={styles.labelStyle}>City: </Text>
                <TextInput
                   style={styles.inputStyle} 
                   placeholder='Select City'
                   autoCorrect={false}
                   value={city}
                   onChangeText={setCity}
                />

            </View>
            <View style={styles.entryStyle}>
                <Text style={styles.labelStyle}>Date: </Text>
                <TextInput
                   style={styles.inputStyle} 
                   placeholder='Select Date'
                   autoCorrect={false}
                   value={date}
                   onChangeText={setDate}
                />

            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    entryStyle: {
        flexDirection: 'row'
    },
    labelStyle: {
        alignSelf: 'center'
    },
    inputStyle: {
        flex: 1
    }
});

export default EventsSearchScreen;