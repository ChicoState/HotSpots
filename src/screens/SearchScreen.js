import React, { useState } from "react";
import { View, Text, StyleSheet, ScrollView } from "react-native";
import SearchBar from "../components/SearchBar";
import useBusinesses from "../hooks/useBusinesses";
import BusinessesList from "../components/BusinessesList";
import getDirections from "react-native-google-maps-directions";

const SearchScreen = function() {
  /*
   * The useState() hook takes the default state as an argument and returns an array
   * with a stateful value (term) and a function to update it (setTerm).
   * The SearchBar component below uses this to update (term)
   * which allows us to pass through user input wherever we need within the project.
   */
  const [term, setTerm] = useState("");
  const [searchApi, businesses, errorMessage] = useBusinesses();

  const filterBusinessesByPrice = function(price) {
    // price === "$" || "$$" || "$$$" || "$$$$"
    return businesses.filter(function(business) {
      return business.price === price;
    });
  };

  return (
    // {flex: 1} constrains the View component so that it doesn't render anything out of view of the screen.
    <View style={{ flex: 1, backgroundColor: "black" }}>
      <SearchBar
        term={term}
        onTermChange={setTerm}
        onTermSubmit={function() {
          searchApi(term);
        }}
      />
      {errorMessage ? <Text>{errorMessage}</Text> : null}
      <ScrollView>
        <BusinessesList businesses={filterBusinessesByPrice("$")} title="$" />
        <BusinessesList businesses={filterBusinessesByPrice("$$")} title="$$" />
        <BusinessesList
          businesses={filterBusinessesByPrice("$$$")}
          title="$$$"
        />
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({});

export default SearchScreen;
