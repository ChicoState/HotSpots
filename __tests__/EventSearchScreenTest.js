import 'react-native';
import React from 'react';
import DealScreen from '../src/screens/DealsScreen';
import SearchScreen from '../src/screens/SearchScreen'
import renderer from 'react-test-renderer';
import act from 'react-test-renderer'
import TestRenderer from 'react-test-renderer'
import EventsSearchScreen from '../src/screens/EventsSearchScreen'


describe( "render tests", () => {
    jest.useFakeTimers();
    it('Eventsearchscreen Test', async done => {
        var snap;  
       await TestRenderer.act(async () => {
            snap = await renderer.create(<EventsSearchScreen />);
       });
       let tree = snap.toJSON;
       await expect(tree).toMatchSnapshot();
        done();
    });
});