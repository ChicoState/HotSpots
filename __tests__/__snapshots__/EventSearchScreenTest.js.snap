// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests Eventsearchscreen Test 1`] = `[Function]`;
